featureflags.go

```diff
@@ -259,17 +264,16 @@ func (poller *FeatureFlagsPoller) computeFlagLocally(flag FeatureFlag, distinctI
 	}
 
 	if flag.Filters.AggregationGroupTypeIndex != nil {
-
+		poller.mutex.RLock()
```

Comment:
Could you walk me through the error not having this leads to?

And as a follow up, do we need something similar for `poller.GetFeatureFlags()` ?

Otherwise looking good, thanks!

Reply:
I went to prove the efficacy of my contributions by making all tests run in parallel via `t.Parallel()`, but the truth is this code is highly concurrent and not race-condition free. When I looked through the parts of the code that would need to be guarded by mutexes to make it concurrency safe, it ended up being a significant refactor, and I've decided I have no reason to care about it if ya'll didn't already. :)

Reply:
to be clear, I'm not asking you to prove it @verygoodsoftwarenotvirus , just asking you what case you're trying to defend against because I'm unsure.

Worth noting here that a stale read is not a problem, and it's favourable to blocking, since it will eventually update anyway.

What panics is when we read before it being defined, and I don't think that's happening anywhere here (if it does, we need to fix!)